test/tape/index.js

```diff
@@ -0,0 +1,62 @@
+import test from 'tape';
+import thunk from 'redux-thunk';
+
+import { assertions, registerMiddlewares } from '../../src';
+import actions from '../testingData/actions';
+
+registerMiddlewares([thunk]);
+
+describe('tape', () => {
+  describe('t.plan', () => {
+    it('should finish the test without hanging', (done) => {
+      test('tape test', t => {
+        t.plan(1);
+
+        assertions.toDispatchActionsWithState(
+          { property: 'value' },
+          actions.actionCreatorWithGetState(),
+          actions.actionWithGetState({ property: 'value' }),
+          () => {
+            t.pass();
+            done();
+          },
+          t.fail
+        );
+      });
+    });
+  });
+
+  describe('t.plan', () => {
+    it('should finish the test without hanging', (done) => {
+      test('tape test', t => {
+        t.plan(1);
+
+        assertions.toDispatchActionsWithState(
+          { property: 'value' },
+          actions.actionCreatorWithGetState(),
+          actions.actionWithGetState({ property: 'value' }),
+          t.pass,
+          t.fail
+        ).then(done);
+      });
+    });
+  });
+
+  describe('t.end', () => {
+    it('should finish the test without hanging', (done) => {
+      test('tape test', t => {
+        assertions.toDispatchActionsWithState(
+          { property: 'value' },
+          actions.actionCreatorWithGetState(),
+          actions.actionWithGetState({ property: 'value' }),
+          () => {
+            t.pass();
+            t.end();
+            done();
+          },
+          t.fail
+        );
+      });
+    });
+  });
+});
```

Comment:
You are having problems because you are mixing mocha with tape, both are testing frameworks with own expectations, etc. Just use tape in those tests and it will work.